In [1]:
#Import libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import matplotlib
import os 

In [3]:
#Import clean data sets 
path = r'C:\Users\ctede\OneDrive\Desktop\World University Rankings Analysis'
#CWUR = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'CWUR_clean.pkl'))
ARWU = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'ARWU_merged.pkl'))
#Times = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'Times_clean.pkl'))

In [4]:
ARWU.head()

world_rank     university_name national_rank  total_score  alumni  award  \
0          1  Harvard University             1        100.0   100.0  100.0   
1          1  Harvard University             1        100.0   100.0  100.0   
2          1  Harvard University             1        100.0   100.0  100.0   
3          1  Harvard University             1        100.0   100.0  100.0   
4          1  Harvard University             1        100.0   100.0  100.0   

    hici     ns    pub   pcp  year        country         region  \
0  100.0  100.0  100.0  72.4  2005  United States  North America   
1  100.0  100.0  100.0  73.6  2006  United States  North America   
2  100.0  100.0  100.0  73.0  2007  United States  North America   
3  100.0  100.0  100.0  74.1  2008  United States  North America   
4  100.0  100.0  100.0  74.8  2009  United States  North America   

  Score Category  
0        Over 65  
1        Over 65  
2        Over 65  
3        Over 65  
4        Over 65

In [5]:
ARWU.dtypes

world_rank          object
university_name     object
national_rank       object
total_score        float64
alumni             float64
award              float64
hici               float64
ns                 float64
pub                float64
pcp                float64
year                 int64
country             object
region              object
Score Category      object
dtype: object

In [13]:
ARWU.isnull().sum()

world_rank            0
university_name       0
national_rank         0
total_score        2299
alumni                0
award                 1
hici                  1
ns                   12
pub                   1
pcp                   1
year                  0
country               0
region                0
Score Category     2299
dtype: int64

In [14]:
ARWU.shape

(3014, 14)

In [21]:
ARWU_filter = ARWU[ARWU['world_rank'].str.contains('-')]

In [22]:
ARWU_filter

world_rank                      university_name national_rank  \
447     101-150              University of Rochester         52-65   
467     101-150      Case Western Reserve University         53-64   
468     101-150      Case Western Reserve University         52-65   
498     101-150                    Tohoku University           5-7   
499     101-150                    Tohoku University           4-6   
...         ...                                  ...           ...   
3009    401-500           Capital Medical University         28-32   
3010    401-500  Queensland University of Technology            20   
3011    401-500      Sharif University of Technology             2   
3012    401-500                  University of Genoa         11-20   
3013    401-500                University of Seville          9-13   

      total_score  alumni  award  hici    ns   pub   pcp  year        country  \
447           NaN    24.6    7.7  25.2  20.8  39.0  20.8  2015  United States   
467           NaN    31.1   10.9  17.9  22.0  39.0  23.0  2014  United States   
468           NaN    30.3   10.9  15.9  19.0  39.7  22.9  2015  United States   
498           NaN    15.3    0.0  21.4  23.2  55.1  20.7  2012          Japan   
499           NaN    14.9    0.0  20.7  21.8  53.2  21.3  2014          Japan   
...           ...     ...    ...   ...   ...   ...   ...   ...            ...   
3009          NaN     0.0    0.0   0.0   0.0  36.0  32.2  2015          China   
3010          NaN     0.0    0.0   0.0   3.3  34.6  18.5  2015      Australia   
3011          NaN    15.4    0.0   3.6   0.0  29.1  15.3  2015           Iran   
3012          NaN     0.0    0.0   7.2   4.2  31.7  17.5  2015          Italy   
3013          NaN     0.0    0.0   0.0   7.5  33.7  11.3  2015          Spain   

             region Score Category  
447   North America            NaN  
467   North America            NaN  
468   North America            NaN  
498            Asia            NaN  
499            Asia            NaN  
...             ...            ...  
3009           Asia            NaN  
3010      Australia            NaN  
3011           Asia            NaN  
3012         Europe            NaN  
3013         Europe            NaN  

[2299 rows x 14 columns]

Do all rows that have a null/missing total score, also have a world rank with a dash "-"? 

In [25]:
ARWU_filter.isnull().head(100)

world_rank  university_name  national_rank  total_score  alumni  award  \
447       False            False          False         True   False  False   
467       False            False          False         True   False  False   
468       False            False          False         True   False  False   
498       False            False          False         True   False  False   
499       False            False          False         True   False  False   
..          ...              ...            ...          ...     ...    ...   
759       False            False          False         True   False  False   
760       False            False          False         True   False  False   
761       False            False          False         True   False  False   
762       False            False          False         True   False  False   
763       False            False          False         True   False  False   

      hici     ns    pub    pcp   year  country  region  Score Category  
447  False  False  False  False  False    False   False            True  
467  False  False  False  False  False    False   False            True  
468  False  False  False  False  False    False   False            True  
498  False  False  False  False  False    False   False            True  
499  False  False  False  False  False    False   False            True  
..     ...    ...    ...    ...    ...      ...     ...             ...  
759  False  False  False  False  False    False   False            True  
760  False  False  False  False  False    False   False            True  
761  False  False  False  False  False    False   False            True  
762  False  False  False  False  False    False   False            True  
763  False  False  False  False  False    False   False            True  

[100 rows x 14 columns]

Most likely, yes. The number of missing values for the total score is 2299, for the score category is 2299, and for the world ranks that are not well-defined 2299.

In [27]:
#remove the missing rows
#df_prods_clean = df_prods[df_prods['product_name'].isnull() == False]
#df_prods_clean
ARWU_no_nulls = ARWU[ARWU['total_score'].isnull() == False]
ARWU_no_nulls

world_rank           university_name national_rank  total_score  alumni  \
0             1        Harvard University             1        100.0   100.0   
1             1        Harvard University             1        100.0   100.0   
2             1        Harvard University             1        100.0   100.0   
3             1        Harvard University             1        100.0   100.0   
4             1        Harvard University             1        100.0   100.0   
...         ...                       ...           ...          ...     ...   
2947         93  Arizona State University            47         24.5     0.0   
2948         96                 KU Leuven             2         24.2     0.0   
2949         90                 KU Leuven             2         24.7     0.0   
3003         16  Johns Hopkins University            14         46.3    37.7   
3004        100      Texas A&M University            51         23.9     0.0   

      award   hici     ns    pub   pcp  year        country         region  \
0     100.0  100.0  100.0  100.0  72.4  2005  United States  North America   
1     100.0  100.0  100.0  100.0  73.6  2006  United States  North America   
2     100.0  100.0  100.0  100.0  73.0  2007  United States  North America   
3     100.0  100.0  100.0  100.0  74.1  2008  United States  North America   
4     100.0  100.0  100.0  100.0  74.8  2009  United States  North America   
...     ...    ...    ...    ...   ...   ...            ...            ...   
2947   20.0   22.2   25.5   42.6  19.1  2015  United States  North America   
2948    0.0   29.5   19.1   54.9  29.1  2014        Belgium         Europe   
2949    0.0   30.1   18.6   56.6  30.4  2015        Belgium         Europe   
3003   33.6   44.0   44.9   70.2  28.8  2015  United States  North America   
3004    0.0   34.3   22.7   49.5  20.9  2015  United States  North America   

     Score Category  
0           Over 65  
1           Over 65  
2           Over 65  
3           Over 65  
4           Over 65  
...             ...  
2947       Under 25  
2948       Under 25  
2949       Under 25  
3003  Btwn 45 to 64  
3004       Under 25  

[715 rows x 14 columns]

In [28]:
#Check filter again 
ARWU_filter_2 = ARWU_no_nulls[ARWU_no_nulls['world_rank'].str.contains('-')]
ARWU_filter_2

Empty DataFrame
Columns: [world_rank, university_name, national_rank, total_score, alumni, award, hici, ns, pub, pcp, year, country, region, Score Category]
Index: []

# Export ARWU without nulls

In [29]:
#Export the CLEAN products dataframe 
ARWU_no_nulls.to_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'ARWU_no_nulls.pkl'))